test/testfolderman.cpp

```diff
@@ -24,7 +26,108 @@ class TestFolderMan: public QObject
+        QScopedPointer<FakeQNAM> fakeQnam(new FakeQNAM({}));
+        OCC::AccountPtr account = OCC::Account::create();
+        account->setCredentials(new FakeCredentials{fakeQnam.data()});
+        account->setUrl(QUrl(("http://example.de")));
+        OCC::AccountManager::instance()->addAccount(account);
+
+        FakeFolder fakeFolder{FileInfo{}};
+        fakeFolder.remoteModifier().mkdir("A");
+
+        fakeFolder.remoteModifier().insert(firstSharePath, 100);
+        const auto firstShare = fakeFolder.remoteModifier().find(firstSharePath);
+        QVERIFY(firstShare);
+        firstShare->permissions.setPermission(OCC::RemotePermissions::IsShared);
+
+        fakeFolder.remoteModifier().mkdir("A/B");
+
+        fakeFolder.remoteModifier().insert(secondSharePath, 100);
+        const auto secondShare = fakeFolder.remoteModifier().find(secondSharePath);
+        QVERIFY(secondShare);
+        secondShare->permissions.setPermission(OCC::RemotePermissions::IsShared);
+
+        FolderMan *folderman = FolderMan::instance();
+        QCOMPARE(folderman, &_fm);
+        OCC::AccountState *accountState = OCC::AccountManager::instance()->accounts().first().data();
+        const auto folder = folderman->addFolder(accountState, folderDefinition(fakeFolder.localPath()));
+        QVERIFY(folder);
+
+        auto realFolder = FolderMan::instance()->folderForPath(fakeFolder.localPath());
+        QVERIFY(realFolder);
+
+        QVERIFY(fakeFolder.syncOnce());
+        QCOMPARE(fakeFolder.currentLocalState(), fakeFolder.currentRemoteState());
+
+        fakeQnam->setOverride([this, accountState, &fakeFolder](QNetworkAccessManager::Operation op, const QNetworkRequest &req, QIODevice *device) {
+            Q_UNUSED(device);
+            QNetworkReply *reply = nullptr;
+
+            if (op != QNetworkAccessManager::DeleteOperation) {
+                reply = new FakeErrorReply(op, req, this, 405);
+                return reply;
+            }
+
+            if (req.url().path().isEmpty()) {
+                reply = new FakeErrorReply(op, req, this, 404);
+                return reply;
+            }
+
+            const auto filePathRelative = req.url().path().remove(accountState->account()->davPath());
+
+            const auto foundFileInRemoteFolder = fakeFolder.remoteModifier().find(filePathRelative);
+
+            if (filePathRelative.isEmpty() || !foundFileInRemoteFolder) {
+                reply = new FakeErrorReply(op, req, this, 404);
+                return reply;
+            }
+
+           fakeFolder.remoteModifier().remove(filePathRelative);
+           reply = new FakePayloadReply(op, req, {}, nullptr);
+
+           emit incomingShareDeleted();
+           
+           return reply;
+        });
+
+        QSignalSpy incomingShareDeletedSignal(this, &TestFolderMan::incomingShareDeleted);
+
+        // verify first share gets deleted
+        folderman->leaveShare(fakeFolder.localPath() + firstSharePath);
+        QTest::qWait(100);
```

Comment:
better do
`QVERIFY(incomingShareDeletedSignal.wait(100));`
that will validate that the signal has been received at least once and will return faster than `100` milliseconds if the signal is emitted before the timeout

Reply:
@mgallien no, the test then fails

Reply:
yes because the signal has been emitted before you call `QTest::qWait(100);` so calling `QVERIFY(incomingShareDeletedSignal.wait(100));` would fail
I would just remove the call to `QTest::qWait` because it is not needed